project/src/test/java/com/google/sps/data/PlacesFetcherTest.java

```diff
@@ -209,4 +209,26 @@ public void fetch_PlaceDetailsQueryFails_throwsFetcherException() throws Excepti
     assertTrue(thrown.getCause() instanceof IOException);
     assertTrue(thrown.getMessage().contains("place details"));
   }
+
+  @Test
+  public void createCuisinesQuery_getsValidCuisines_returnsQuery() throws Exception {
+    assertEquals(
+      "sushi|burger|hamburger", placesFetcher.createCuisinesQuery(CUISINES));
+  }
+
+  @Test
+  public void createCuisinesQuery_getsAnEmptyListOfCuisines_returnsEmptyQuery() throws Exception {
+    assertEquals("", placesFetcher.createCuisinesQuery(ImmutableList.of()));
+  }
+
+  @Test
+  public void createCuisinesQuery_getsInvalidCuisines_throwsFetcherException() throws Exception {
+    FetcherException thrown =
+        assertThrows(
+            FetcherException.class,
+            () -> placesFetcher.createCuisinesQuery(ImmutableList.of("blah")));
+    assertTrue(thrown.getCause() instanceof NullPointerException);
```

Comment:
Shouldn't this be a FetcherException?

Reply:
The exception thrown is a FetcherException. Each FetcherException has a cause, so in this case of throwing fetcherException due to invalid cuisine the cause of the exception will be NullPointerException. Do you think I should change the cause to some more general exception?

Reply:
Ah my bad, missed that part.